qiskit/transpiler/passes/layout/sabre_layout.py

```diff
@@ -215,7 +217,80 @@ def run(self, dag):
             self.property_set["layout"] = initial_layout
             self.routing_pass.fake_run = False
             return dag
-        dist_matrix = self.coupling_map.distance_matrix
+        # Combnined
+        layout_components = disjoint_utils.run_pass_over_connected_components(
+            dag, self.coupling_map, self._inner_run
+        )
+        initial_layout_dict = {}
+        final_layout_dict = {}
+        for (
+            layout_dict,
+            final_dict,
+            component_map,
+            _gate_order,
+            _swap_map,
+            _dag,
+        ) in layout_components:
+            initial_layout_dict.update({k: component_map[v] for k, v in layout_dict.items()})
+            final_layout_dict.update({component_map[k]: component_map[v] for k, v in final_dict})
+        self.property_set["layout"] = Layout(initial_layout_dict)
+        # If skip_routing is set then return the layout in the property set
+        # and throwaway the extra work we did to compute the swap map
+        if self.skip_routing:
+            return dag
+        # After this point the pass is no longer an analysis pass and the
+        # output circuit returned is transformed with the layout applied
+        # and swaps inserted
+        dag = self._apply_layout_no_pass_manager(dag)
+        mapped_dag = dag.copy_empty_like()
+        self.property_set["final_layout"] = Layout(
+            {dag.qubits[k]: v for (k, v) in final_layout_dict.items()}
+        )
+        canonical_register = dag.qregs["q"]
+        qubit_indices = {bit: idx for idx, bit in enumerate(canonical_register)}
+        original_layout = NLayout.generate_trivial_layout(self.coupling_map.size())
+        canonical_register = dag.qregs["q"]
+        for (
+            _layout_dict,
+            _final_layout_dict,
+            component_map,
+            gate_order,
+            swap_map,
+            local_dag,
+        ) in layout_components:
```

Comment:
Talking with @kevinhartman offline about this and writing extra test cases to confirm there is a potential bug with classical bits here that will result in a reordering of the data dependency between components. Basically the component which is first in `layout_components` will end up being first on the classical bits regardless of what the actual data dependency is (the order here is from largest component to smallest).

I think the best fix for this is to just skip routing if `layout_components > 1` and just return the layout as metadata. The problem with trying to do routing across components like this is getting the topological order exactly right between components manually is very tricky, especially after the routing pass has done 3 different topological iterations over the subcircuits. It's just easier to run routing again even at a small performance penalty.

Reply:
I fixed this in: https://github.com/Qiskit/qiskit-terra/pull/9802/commits/1575b8ad123f0353a7474b3ab32e2613a0498a04 so we don't run routing as part of `SabreLayout` if there is a potential data dependency between components for classical bits.